flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -170,6 +171,8 @@
 
 	private final Map<ResourceID, Tuple2<TaskManagerLocation, TaskExecutorGateway>> registeredTaskManagers;
 
+	private final ShuffleMaster<?> shuffleMaster;
```

Comment:
Why do we store the `ShuffleMaster` here? Wouldn't it be enough to pass it on to the `createScheduler` method?

Reply:
`createScheduler` is currently called not only from the constructor